sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -1764,6 +1764,68 @@ class Dataset[T] private[sql](
   }
 
   /**
+   * Returns a new Dataset containing union of rows in this Dataset and another Dataset.
+   *
+   * To do a SQL-style set union (that does deduplication of elements), use this function
```

Comment:
Also add a comment `This is different from both `UNION ALL` and `UNION DISTINCT` in SQL. `